packages/sitecore-jss-nextjs/src/utils/utils.ts

```diff
@@ -8,12 +8,14 @@ import { isEditorActive, resetEditorChromes } from '@sitecore-jss/sitecore-jss/u
  * VERCEL_URL is provided by Vercel in case if we are in Preview deployment (deployment based on the custom branch),
  * preview deployment has unique url, we don't know exact url.
  * Similarly, DEPLOY_URL is provided by Netlify and would give us the deploy URL
+ * In production non-editing environments it is desirable to use relative urls, so setting PUBLIC_URL = '' will take precendece
  */
 export const getPublicUrl = (): string => {
+  let url = process.env.PUBLIC_URL;
+  if (url === '') return url;
```

Comment:
_Any_ defined PUBLIC_URL environment variable should take precedence over Vercel/Netlify environment variables (not just empty). Only if PUBLIC_URL is undefined should the Vercel/Netlify logic (and then fallback to localhost:3000) occur.